README.md

```diff
@@ -1,126 +1,258 @@
+If you don't have the Leap repo cloned to your computer yet, [open a terminal](https://itsfoss.com/open-terminal-ubuntu) and navigate to the folder where you want to clone the Leap repository:
+```bash
+cd ~/Downloads
+```
+Clone Leap using either HTTPS:
+```bash
+git clone --recursive https://github.com/AntelopeIO/leap.git
+```
+or SSH:
+```bash
+git clone --recursive git@github.com:AntelopeIO/leap.git
+```
 
-The binary package will be produced in the Leap build directory that was supplied.
+> ℹ️ **HTTPS or SSH git clone** ℹ️  
+Both an HTTPS or SSH git clone will yield the same result: a folder named `leap` containg our source. It doesn't matter which type you use (unless you need to push to remote).
 
-#### Manual (non "pinned") Build Instructions
+Navigate into that folder:
+```bash
+cd leap
+```
 
-<details>
-  <summary>Ubuntu 20.04 & 22.04 Build Instructions</summary>
+### Step 2. Select release tag or branch
 
-Install required dependencies: 
+Choose which [release tag](https://github.com/AntelopeIO/leap/releases) or [branch](#branches) you would like to build, then check it out. If you are not sure, use the [latest release](https://github.com/AntelopeIO/leap/releases/latest). For example, if you want to build release 3.1.2 then you would check it out using its tag, `v3.1.2`. In the example below, replace `v0.0.0` with your selected release tag accordingly:
+```bash
+git fetch --all --tags
+git checkout v0.0.0
 ```
-apt-get update && apt-get install   \
-        build-essential             \
-        cmake                       \
-        curl                        \
-        git                         \
-        libboost-all-dev            \
-        libcurl4-openssl-dev        \
-        libgmp-dev                  \
-        libssl-dev                  \
-        libusb-1.0-0-dev            \
-        llvm-11-dev                 \
-        pkg-config
+
+Once you are on the branch or release tag you want to build, make sure everything is up-to-date:
+```bash
+git pull
+git submodule update --init --recursive
+```
+
+#### Jobs Flag
+⚠️ **A Warning On Parallel Compilation Jobs (`-j` flag)** ⚠️  
+When building C/C++ software, often the build is performed in parallel via a command such as `make -j "$(nproc)"` which uses all available CPU threads. However, be aware that some compilation units (`*.cpp` files) in Leap will consume nearly 4GB of memory to compile. Failures due to memory exhaustion will typically, but not always, manifest as compiler crashes. Using all available CPU threads may also prevent you from doing other things on your computer during compilation. For these reasons, consider adjusting parallelization.
+
+### Step 3. Build the source code
+Select build instructions below for a [Pinned Build](#pinned-build) (preferred) or an [Unpinned Build](#unpinned-build).
+
+ℹ️ **Pinned Build vs. Unpinned Build** ℹ️  
```

Comment:
Same here, might want to make this a quote too by prepending this line with ">".

Reply:
Done.